pkg/kudoctl/util/repo/repo_operator.go

```diff
@@ -75,5 +84,39 @@ func (c *Client) DownloadIndexFile() (*IndexFile, error) {
 	}
 
 	indexFile, err := ParseIndexFile(indexBytes)
+	//TODO (kensipe): err handling such that error in includes are ignored (but not root)
+	//TODO (kensipe): track which includes have happened so there are no repeats
+	for _, include := range indexFile.Includes {
+		iURL, err := url.Parse(include)
+		if err != nil {
+			clog.Printf("Unable to parse include url for %s", include)
+		}
+		nextIndex, err := c.downloadIndexFile(indexFile, iURL)
+		if err != nil {
+			return nil, err
+		}
+		if parent != nil {
+			c.Merge(parent, nextIndex)
+		} else {
+			c.Merge(indexFile, nextIndex)
+		}
```

Comment:
I think this block works incorrectly for nested includes.

Assuming we have three repos that have nested includes: AA includes BB, BB includes CC. 
BB and CC contain operator X.

The final index should contain BB.X, if I understand correctly.

With this code, we would:
- downloadIndexFile(nil, urlAA) -> indexFile for AA has no operator X
- downloadIndexFile(indexFileAA, urlBB)
- downloadIndexFile(indexFileAA, urlCC)
- c.Merge(indexFileAA, indexCC) -> indexFileAA now has indexCC.X operator
- c.Merge(indexFileAA, indexBB) -> X operator is already in there and is skipped.

Either we:
- c.Merge the current index file before we handle the includes
- Not pass the parent IndexFile to `downloadIndexFile` and merge the returned file from that call

Reply:
IF AA -> BB and BB -> CC and BB and CC have X I would expect:

```
downloadIndexFile(nil, urlAA) -> indexFile for AA has no operator X
downloadIndexFile(indexFileAA, urlBB)
downloadIndexFile(indexFileBB, urlCC)
c.Merge(indexFileBB, indexCC) -> indexFileBB would ignore X from CC (it has one)
c.Merge(indexFileAA, indexBB) -> X is merged into AA from BB
```

Reply:
the case where 
```
downloadIndexFile(indexFileAA, urlBB)
downloadIndexFile(indexFileAA, urlCC)
```
in your example would be true if AA included BB and CC